packages/ai-native/src/browser/mcp/tools/grepSearch.ts

```
@@ -0,0 +1,121 @@
+      const results: string[] = [];
+      const files: string[] = [];
+      for (const [fileUri, result] of this.searchService.searchResults.entries()) {
+        results.push(
+          `File: ${fileUri}\n${result
+            .reduce((acc, r) => {
+              if (acc.find((a) => a.line === r.line)) {
+                return acc;
+              }
+              return [...acc, r];
+            }, [] as ContentSearchResult[])
+            .map((r) => `Line: ${r.line}\nContent: ${r.lineText || r.renderLineText}`)
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)